packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -187,58 +211,73 @@ public ResponseEntity<TaskResponse> createConfigureModelTask(
 		final Schema.Permission permission =
 				projectService.checkPermissionCanRead(currentUserService.get().getId(), projectId);
 
-		try {
+		// Grab the document
+		final Optional<DocumentAsset> document = documentAssetService.getAsset(documentId, permission);
+		if (document.isEmpty()) {
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("document.not-found"));
+		}
 
```

Comment:
Why the `try/catch` is removed? the `getAsset` take cares of it?

Reply:
I guess... getAsset does not throw exceptions. However, that is another task I want to go through. Services should throw better errors as well. But that is not in the scope of this initial work.